src/main.rs

```diff
@@ -17,29 +17,43 @@
  along with Blockstack. If not, see <http://www.gnu.org/licenses/>.
 */
 
+#![allow(unused_imports)]
+#![allow(unused_assignments)]
+#![allow(unused_variables)]
+#![allow(dead_code)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+#![allow(non_upper_case_globals)]
```

Comment:
This seems kind of aggressive --

While I appreciate that many of these warnings are annoying, some of them are simply Rust stylistic conventions (camel case, snake case, upper case globals) that there seems to be no real reason to ignore.

dead_code, unused assignments, and unused variables are all things that we should avoid simply so that our code is easier to read. unused assignments and variables, in particular, can obfuscate the real purpose of functions. unused imports seem harmless enough that I'd be perfectly happy turning that warning off.

Reply:
So, the reason I turned them off is because there can be literally hundreds of these warnings, and for a long time they were hiding the one or two warnings about possible integer overflows (a far more severe problem).

* I used to spend an annoyingly large amount of my time trying to get all the "unused import" warnings to go away, despite the fact that an unused import will be ignored by the compiler and linker anyway.

* I don't use snake case in some cases because the function name contains an acronym (e.g. `ECVRF`), which I think is more descriptive.

* I don't use upper-case globals in one case (network message type constants) because they're meant to reflect the same names in an associated `enum`.  For example, `StacksMessageID::Handshake` is a global, and it corresponds to the `enum` type variant `StacksMessageType::Handshake(HandshakeData)`.

* I disable `dead_code` because a lot of the time, I'll disable methods temporarily in order to debug something.

Happy to reverse these now, but I wanted you to know that I wasn't disabling them with the intent of ignoring them outright.